packages/fxa-react/lib/hooks.stories.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback } from 'react';
+import { storiesOf } from '@storybook/react';
+import { useAwait, PromiseState } from './hooks';
+import LoadingSpinner from '../components/LoadingSpinner';
+
+const API_URL = '//worldtimeapi.org/api/ip';
+
+storiesOf('hooks/useAwait', module)
+  .add('basic', () => (
+    <>
+      <UseAwaitExample />
+      <UseAwaitExample apiUrl="//worldtimeapi.org/api/timezone/Europe/London" />
+      <UseAwaitExample apiUrl="//worldtimeapi.org/api/timezone/America/Detroit" />
+    </>
+  ))
+  .add('with initial state', () => (
+    <UseAwaitExample
+      initialState={{
+        result: initialResult(),
+        pending: false,
+        error: undefined,
+      }}
+    />
+  ))
+  .add('with immediate execution', () => (
+    <UseAwaitExample
+      executeImmediately={true}
+      fetchApiFn={() => fetchApi(API_URL)}
+    />
+  ))
+  .add('with error', () => (
+    <UseAwaitExample apiUrl="//worldtimeapi.org/badPage" />
+  ));
+
+const UseAwaitExample = ({
+  fetchApiFn = fetchApi,
+  apiUrl = API_URL,
+  style = {},
+  executeImmediately = undefined as boolean | undefined,
+  initialState = undefined as PromiseState<ApiResult, any> | undefined,
+}) => {
+  const [apiState, apiExecute, apiReset] = useAwait(fetchApiFn, {
+    executeImmediately,
+    initialState,
+  });
+  const executeWithApiUrl = useCallback(() => apiExecute(apiUrl), [
+    apiUrl,
+    apiExecute,
+  ]);
+
+  return (
+    <div style={{ ...containerStyle, ...style }}>
+      <button style={buttonStyle} onClick={apiReset}>
+        Reset
+      </button>
+      <button style={buttonStyle} onClick={executeWithApiUrl}>
+        Request
+      </button>
+      {apiUrl}
+      <ApiStateDisplay {...apiState} />
+    </div>
+  );
+};
+
+const ApiStateDisplay = ({
+  pending,
+  error,
+  result,
+}: PromiseState<ApiResult, any>) => {
+  if (!pending && !error && !result) {
+    return <p>Reset state</p>;
+  } else if (pending) {
+    return <LoadingSpinner />;
```

Comment:
Hm, the loader isn't looking right, but I'm seeing that on the `LoadingSpinner` Storybook as well so nothing to do with implementation here.

Reply:
Huh. That it looks correct when being used in fxa-settings, but not in fxa-react, tells me something's up with Tailwind. [I made an issue.](https://jira.mozilla.com/browse/FXA-2152)